DFC.App.JobProfileTasks.MessageFunctionApp/Services/MessagePreProcessor.cs

```diff
@@ -0,0 +1,107 @@
+                            await messageProcessor.DeleteLocation(jobProfileLocationDeleteServiceBusModel.JobProfileId, jobProfileLocationDeleteServiceBusModel.Id).ConfigureAwait(false);
+                            break;
+                        case MessageContentType.Environment:
+                            var jobProfileEnvironmentDeleteServiceBusModel = JsonConvert.DeserializeObject<JobProfileEnvironmentDeleteServiceBusModel>(message);
+                            await messageProcessor.DeleteEnvironment(jobProfileEnvironmentDeleteServiceBusModel.JobProfileId, jobProfileEnvironmentDeleteServiceBusModel.Id).ConfigureAwait(false);
+                            break;
+                    }
+
+                    break;
+                case MessageActionType.Published:
+                    switch (messageContentType)
+                    {
+                        case MessageContentType.JobProfile:
+                            var jobProfileServiceBusModel = JsonConvert.DeserializeObject<JobProfileServiceBusModel>(message);
+
+                            if (jobProfileServiceBusModel == null)
+                            {
+                                throw new InvalidOperationException($"Service bus model is null");
+                            }
+
+                            await messageProcessor.Save(jobProfileServiceBusModel, messageContentType, sequenceNumber).ConfigureAwait(false);
+                            break;
+                        case MessageContentType.Uniform:
+                            var uniformPatchServiceBusModel = JsonConvert.DeserializeObject<JobProfileUniformPatchServiceBusModel>(messageBody);
+                            var patchUniformModel = mapper.Map<PatchUniformModel>(uniformPatchServiceBusModel);
+                            await messageProcessor.PatchUniform(patchUniformModel).ConfigureAwait(false);
+                            break;
+                        case MessageContentType.Location:
+                            var locationPatchServiceBusModel = JsonConvert.DeserializeObject<JobProfileLocationPatchServiceBusModel>(messageBody);
+                            var patchLocationModel = mapper.Map<PatchLocationModel>(locationPatchServiceBusModel);
+                            await messageProcessor.PatchLocation(patchLocationModel).ConfigureAwait(false);
+                            break;
+                        case MessageContentType.Environment:
+                            var jobProfileEnvironmentPatchServiceBusModel = JsonConvert.DeserializeObject<JobProfileEnvironmentPatchServiceBusModel>(messageBody);
+                            var patchEnvironmentsModel = mapper.Map<PatchEnvironmentsModel>(jobProfileEnvironmentPatchServiceBusModel);
+                            await messageProcessor.PatchEnvironment(patchEnvironmentsModel).ConfigureAwait(false);
+                            break;
+                    }
+
+                    break;
+            }
+        }
+    }
```

Comment:
This is very large method, need to be split down. Am not too sure about nested switches are helping either.